tests/unit/experimental/globus_app/test_globus_app.py

```diff
@@ -72,28 +73,27 @@ def test_user_app_native():
 
 
 def test_user_app_login_client():
-    mock_client = mock.Mock(environment="production")
+    mock_client = mock.Mock(client_id="mock-client_id", environment="production")
     user_app = UserApp("test-app", login_client=mock_client)
 
     assert user_app.app_name == "test-app"
     assert user_app._login_client == mock_client
+    assert user_app.client_id == "mock-client_id"
 
 
 def test_user_app_no_client_or_id():
-    with pytest.raises(GlobusSDKUsageError) as exc:
+    msg = (
+        "Could not to set up a globus login client. One of client_id or "
+        "login_client is required."
```

Comment:
I didn't immediately see where this was raised in the code, but "Could not to set up" needs to be addressed in the original location.

```suggestion
        "Could not set up a Globus login client. "
        "One of client_id or login_client is required."
```